packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -29,6 +30,10 @@ if (process.env.NODE_ENV === 'development') {
 }
 
 const globalStylesheetsParentElement: Element = document.head || document.body || document;
+const supportsConstructableStyleSheets = isFunction((CSSStyleSheet.prototype as any).replaceSync);
+const styleElements: { [content: string]: HTMLStyleElement } = create(null);
+const styleSheets: { [content: string]: CSSStyleSheet } = create(null);
+const nodesToStyleSheets = new WeakMap<Node, { [content: string]: true }>();
```

Comment:
The `styleElements` and `styleSheets` are technically a memory leak, but I don't think it's a big deal in practice. With synthetic shadow, we already don't clean up `<style>` tags added to the `<head>`, so we're already leaking there.

I would also prefer not to maintain a mapping of `string` -> `CSSStyleSheet` at all, but that would require modifying the `style-compiler`, and I made a point of not doing that because I tried that in #2290 and it adds a lot of complexity.